pkg/controller/replicaset/replica_set_utils.go

```diff
@@ -115,10 +125,23 @@ func calculateStatus(rs *apps.ReplicaSet, filteredPods []*v1.Pod, manageReplicas
 			reason = "FailedDelete"
 		}
 		cond := NewReplicaSetCondition(apps.ReplicaSetReplicaFailure, v1.ConditionTrue, reason, manageReplicasErr.Error())
```

Comment:
do we need this logic for backwards compatibility? That is acknowledging LastUpdateTime depending on features.ExtendedWorkloadConditions feature gate?

If yes we could also move this check to `SetCondition` function.